src/cargo/sources/registry/index.rs

```diff
@@ -148,12 +150,7 @@ impl<'cfg> RegistryIndex<'cfg> {
             features,
             yanked,
             links,
-        } = super::DEFAULT_ID.with(|slot| {
-            *slot.borrow_mut() = Some(self.source_id.clone());
-            let res = serde_json::from_str::<RegistryPackage>(line);
-            drop(slot.borrow_mut().take());
-            res
-        })?;
+        } = RegistryPackageDefault(self.source_id.clone()).deserialize(&mut json_de)?;
```

Comment:
I don't know what SourceId is but this can be `RegistryPackageDefault(&self.source_id)` if clone is slow.